src/Columns/IColumnImpl.h

```diff
@@ -100,15 +101,8 @@ void IColumn::compareImpl(const Derived & rhs, size_t rhs_row_num,
         if constexpr (use_indexes)
             row = indexes[i];
 
-        int res = compareAt(row, rhs_row_num, rhs, nan_direction_hint);
-
-        /// We need to convert int to Int8. Sometimes comparison return values which do not fit in one byte.
-        if (res < 0)
-            compare_results[row] = -1;
-        else if (res > 0)
-            compare_results[row] = 1;
-        else
-            compare_results[row] = 0;
+        int res = static_cast<const Derived *>(this)->compareAt(row, rhs_row_num, rhs, nan_direction_hint);
+        compare_results[row] = static_cast<Int8>(res);
```

Comment:
Column compare must return 1, 0, -1, otherwise it is logical error, also direction must be -1, 1. If this invariant is broken we need to fix it, without introducing workaround here.

Reply:
I absolutely agree with your point but that comment looks suspicious. It sounds like it's author knew cases when the invariant was broken.
Maybe there should be a debug assert to this invariant?